tgui-next/packages/tgui/interfaces/CodexGigas.js

```diff
@@ -0,0 +1,202 @@
+import { act } from '../byond';
+import { Button, LabeledList, Section } from '../components';
+
+// TODO: refactor the backend of this it's a trainwreck
+// new ui devs do not copypaste or use it as reference it sucks
+export const CodexGigas = props => {
+  const { state } = props;
+  const { config, data } = state;
+  const { ref } = config;
+  return (
+    <Section>
+      {data.name}
+      <LabeledList>
+        <LabeledList.Item label="Prefix">
+          <Button
+            content="Dark"
+            disabled={data.currentSection !== 1}
+            onClick={() => act(ref, 'Dark ')}
+          />
+          <Button
+            content="Hellish"
```

Comment:
i honestly dont know js well enough but can you not do like a for loop that iterates through a list that has all these? like ["Dark","Hellish","Fallen",...]
this is just so ugly

Reply:
You could, but the backend would need to be reworked a lot. I don't really want to have to do that right now.

Reply:
Wait, no, I see what you mean. I can do that yeah.